token/common.tact

```diff
@@ -93,3 +93,33 @@ struct BurnNotification {
 
     @derive impl Serialize {}
 }
+
+contract interface JettonOwner {
+    msg receive_notification(prefix: 0x7362d09c, notification: TransferNotification);
+}
+
+contract interface ResponseReceiver {
+    msg receive_response(prefix: 0xd53276db, response: Excesses);
+}
+
+contract interface Master {
+    msg receive_burn_notification(prefix: 0xd53276db, notification: BurnNotification);
+}
+
+contract interface Jetton {
+    var balance: Coins;
+    var owner: JettonOwner;
+    var master: Master;
+    var wallet_code: Cell;
```

Comment:
A thought: if `Ref[T]` means the same as `^` in TL-B, then composition of types such as `Cell=Ref[Any]` creates more words that mean `^`. Other types may wrap around Cell or Ref and make it hard to read the structure definition: you would have hard time counting where do reference lurk and whether you have enough room for them.

In TL-B you may also hide things like that, but normally would put complex types under ^ immediately instead of inlining them. Maybe we could have a convention to not newtype Ref[T] apart from Cell and use explicit Ref[X] for complex types as in TL-B?

Reply:
Meta-thought: writing `Ref[MyType]` could look cumbersome and heavyweight compared to `^MyType`. Like you are wasting gas and nudge you to reshuffle your data to minimize usage of Ref. Maybe that's a good thing.